pkg/resources/vfioResource.go

```diff
@@ -52,11 +54,11 @@ func (rp *vfioResource) GetDeviceSpecs(pciAddr string) []*pluginapi.DeviceSpec {
 
 	vfioDev, err := utils.GetVFIODeviceFile(pciAddr)
 	if err != nil {
-		glog.Errorf("GetDeviceSpecs(): error getting vfio device file for device: %s", pciAddr)
+		glog.Errorf("GetDeviceSpecs(): error getting vfio device file for device: %s, %s", pciAddr, err.Error())
 	} else {
 		devSpecs = append(devSpecs, &pluginapi.DeviceSpec{
 			HostPath:      vfioDev,
-			ContainerPath: vfioDev,
+			ContainerPath: strings.Replace(vfioDev, "noiommu-", "", 1),
```

Comment:
This is indeed clumsy code.  I wanted to minimize changes to the current code as much as possible.  I could change the return signature of the function call to return a hostpath and a container path.  

The reason for this Replace is that vfioDev might contain. noiommu-0(1,2,etc) in the VM use case.

Reply:
> I could change the return signature of the function call to return a hostpath and a container path
this will work as well, then all logic is rolled into `GetVFIODeviceFile(...)`